nflow-engine/src/main/java/io/nflow/engine/internal/guice/EngineModule.java

```diff
@@ -0,0 +1,176 @@
+    install(new EngineInitModule());
+  }
+
+  @Provides
+  @Inject
+  @NFlow
+  @Singleton
+  public AbstractResource nflowNonSpringWorkflowsListing(Environment env) {
+    return engineConfiguration.nflowNonSpringWorkflowsListing(env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public ThreadFactory nflowThreadFactory() {
+    return engineConfiguration.nflowThreadFactory();
+  }
+
+  @Provides
+  @Inject
+  @Singleton
+  public WorkflowInstanceExecutor nflowExecutor(@NFlow ThreadFactory factory, Environment env) {
+    return engineConfiguration.nflowExecutor(factory, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public ObjectMapper nflowObjectMapper() {
+    return engineConfiguration.nflowObjectMapper();
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public DataSource nflowDataSource(Environment env) {
+    return getDatabaseConfiguration(env).nflowDatasource(env, metricRegistry);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public DatabaseInitializer nflowDatabaseInitializer(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowDatabaseInitializer(dataSource, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public JdbcTemplate nflowJdbcTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public NamedParameterJdbcTemplate nflowNamedParameterJdbcTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowNamedParameterJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public TransactionTemplate nflowTransactionTemplate(@NFlow DataSource dataSource, Environment env) {
+    return getDatabaseConfiguration(env).nflowTransactionTemplate(new DataSourceTransactionManager(dataSource));
+  }
+
+  @Provides
+  @Singleton
+  @Inject
+  public SQLVariants nflowSQLVariants(Environment env) {
+    return getDatabaseConfiguration(env).sqlVariants();
+  }
+
+  private DatabaseConfiguration getDatabaseConfiguration(Environment env) {
+    DatabaseConfiguration db;
+    String dbtype = env.getProperty("nflow.db.type", String.class);
+    switch (dbtype) {
+    case "h2":
+      db = new H2DatabaseConfiguration();
+      break;
+    case "mysql":
+      db = new MysqlDatabaseConfiguration();
+      break;
+    case "oracle":
+      db = new OracleDatabaseConfiguration();
+      break;
+    case "postgresql":
+      db = new PgDatabaseConfiguration();
+      break;
+    default:
+      throw new RuntimeException("Unknown DB");
+    }
+    return db;
+  }
+
+  class EngineInitModule extends AbstractModule {
+    @Override
+    protected void configure() {
+      requestInjection(this);
+    }
+
+    @Inject
+    void initPostConstruct(WorkflowDefinitionService workflowDefinitionService) throws Exception {
+      workflowDefinitionService.postProcessWorkflowDefinitions();
```

Comment:
I think this is not needed because the `WorkflowLifecycle` injected to `initLifeCycleAutoStart` below will run the `postProcessWorkflowDefinitions` automatically when requested.